examples/lock-app/cyw30739/src/LockManager.cpp

```diff
@@ -275,58 +300,77 @@ bool LockManager::SetUser(uint16_t userIndex, chip::FabricIndex creator, chip::F
 
     for (size_t i = 0; i < totalCredentials; ++i)
     {
-        mCredentials[i]                 = credentials[i];
-        mCredentials[i].CredentialType  = 1;
-        mCredentials[i].CredentialIndex = i + 1;
+        mCredentials[adjustedUserIndex][i]                 = credentials[i];
+        mCredentials[adjustedUserIndex][i].CredentialType  = 1;
+        mCredentials[adjustedUserIndex][i].CredentialIndex = i + 1;
     }
 
-    userInStorage.credentials = chip::Span<const DlCredential>(mCredentials.Get(), totalCredentials);
+    userInStorage.credentials = chip::Span<const DlCredential>(mCredentials[adjustedUserIndex].Get(), totalCredentials);
 
     // Save user information in NVM flash
-    CYW30739Config::WriteConfigValueBin(CYW30739Config::kConfigKey_LockUser, reinterpret_cast<const uint8_t *>(&userInStorage),
-                                        sizeof(EmberAfPluginDoorLockUserInfo));
+    CYW30739Config::WriteConfigValueBin(CYW30739Config::kConfigKey_LockUser, reinterpret_cast<const uint8_t *>(&mLockUsers),
+                                     sizeof(EmberAfPluginDoorLockUserInfo) * mMaxUsers);
 
     CYW30739Config::WriteConfigValueBin(CYW30739Config::kConfigKey_UserCredentials,
-                                        reinterpret_cast<const uint8_t *>(mCredentials.Get()), sizeof(DlCredential));
+                                     reinterpret_cast<const uint8_t *>(mCredentials[adjustedUserIndex].Get()),
+                                     sizeof(DlCredential) * totalCredentials);
```

Comment:
This does not make sense to me.  This is writing to the single kConfigKey_UserCredentials key, but writing out the credentials for one user?